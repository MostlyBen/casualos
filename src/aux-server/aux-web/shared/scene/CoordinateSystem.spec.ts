import { Vector3 } from '@casual-simulation/three';
import { cartesianToLatLon, latLonToCartesian } from './CoordinateSystem';

describe('CoordinateSystem', () => {
    describe('latLonToCartesian()', () => {
        it('should return a Vector3 for the given latitude and longitude and altitude', () => {
            let input = new Vector3(10, 10, 15);
            let output = latLonToCartesian(1, input);
            expect(output).toMatchSnapshot();
        });
    });

    describe('cartesianToLatLon()', () => {
        it('should return a Vector3 for the given latitude and longitude and altitude', () => {
            let input = new Vector3(
                15.517540966287266,
                2.7361611466053493,
                2.7783708426708853
            );
            let output = cartesianToLatLon(1, input);

            expect(output.x).toBeCloseTo(10);
            expect(output.y).toBeCloseTo(10);
            expect(output.z).toBeCloseTo(15);
        });
    });
});
