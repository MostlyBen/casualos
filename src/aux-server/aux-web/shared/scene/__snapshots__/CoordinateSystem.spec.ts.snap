// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoordinateSystem latLonToCartesian() should return a Vector3 for the given latitude and longitude and altitude 1`] = `
Vector3 {
  "x": 15.517540966287266,
  "y": 2.7361611466053493,
  "z": 2.7783708426708853,
}
`;
